packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -1074,6 +1074,8 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     } else {
                         reusedBlobs++;
                         id = `${this.lastSummaryHandle}${completePath}`;
+                        const updatedPath = `/.app${path}/${key}`;
```

Comment:
I do not think this is sufficient.
The blobs that already exist in snapshot we loaded from needs to be reused too.
Today they are, but only in N+1 snapshot. If they were not used in N+1 snapshot, then we wipe out information about such blobs and it's not available when generating snapshot N+2.
We need to fix it, but we also need to be careful to "forget" about all the blobs that are gone in any of the snapshots (i.e. part of the tree is deleted).

Reply:
yes, spo folks advise us to reuse the blobs only from last summary.

Reply:
Actually, here we keep info of all paths that are used in current summary. So any of the path that does not exists anymore, will not present in latest cache.

Reply:
I do not think this is correct. They are present in last summary, we just get it by reusing whole sub-tree.
The value of this fix without full fix is simply not interesting. We should assume that content is built slowly from summary to summary (most cases), and failure to reuse blobs that were created over many summaries would result in pure reuse.
I believe referencing such blobs would work just fine. If not, we should escalate it to SPO, as that's the core of reuse.

Reply:
So 1 question is how do we know currently which blobs are not referenced any more and that we should clear those from the cache in case we start preserving blobs from beginning of document?

Reply:
We should do comparison of blob paths against paths in current summary. If path does not exist in current summary (here I mean none of the nodes is reused from previous summary, and new summary does not contain such path), or if same path has different content, then we have to forget about prior blob. Otherwise we know it will be (under same path) in summary we are generating, so we need to keep knowledge about it.

Please note that even that is not enough (long term). We can have blob (reused) under two different paths, but we remember only one path. When blob under that path changes, forgetting about it ensures we do not reference some other different blob. But if we could realize that blob somewhere still has same sha, then we would be able to reuse it.
It will matter a lot once we allow deleting components (via GC), as  it's possible that there are a lot of same blos out there, but we just chose the "wrong" path to pivot.

Maybe something like that:
we keep path -> sha, and sha -> path mappings.
We recalculate the first map using new data (and keeping old data if it did not change). And we always rebuild the second map using first one from scratch.